code/software/libs/src/machine/machine.S

```diff
@@ -99,3 +99,12 @@ mcHalt::
     stop    #$2700
     bra.s   mcHalt
 
+; Call ROM reset vector to "soft reset" the machine

```

Comment:
I suspect this might cause problems with the 9958, which is quite particular about the order things happen in, but I think this is a good addition, and that should probably be fixed in the bootstrap code, maybe with a warm vs cold vector.

Given that, is it worth doing `move.l $4,A0` here? The current firmware should copy the ROM start address to that address at startup, and warm reboot could be 'fixed' by changing the address to a warm boot vector once enough init is done...

Reply:
Okay, I fully understand.  I have removed all "evidence" of the soft-reset (for now, but I do like your vector idea, perhaps using a "check" word somewhere to decide cold/warm boot - which you may already have).  I verified it still returns to loader nicely without it (I think it didn't at one point - or perhaps "noreturn" on other examples made me "worry" and led to my "tactical nuclear option" - but likely "my bad" or something).  I also added a tiny bit of code that on exit will disable GPIO interrupt and restore the vector (neither seemed needed - but seemed good "example hygiene").

Reply:
One other thought.  For my SPI loader  and similar utilities to be (more) "useful", ideally there would be a way to set an interrupt vector and have it stay set (and enabled) after running other programs and returning to loader.  Of course in the fullness of time it (or something similar) will likely be in my ROMs, but some kind of "resident program" mechanism seems useful - if only for developing this and similar utilities (either with them "hiding" in low-meory or maybe even "reserving" some high memory - dare I say "TSR style" :sweat_smile: ).   Seemed kind of related, so leaving comment here.

Reply:
Cool. Yes I think we should add the concept of warm/cold reboots, and at that point look at adding mcReset back in to soft reset. I still kinda feel like we can just switch out the address in RAM at $4 once enough setup is done so that warm reboot makes sense, and have mcReset just jump to whatever is pointed to by $4. 

Completely agree on the ability to keep stuff resident - I think implementing warm reboot makes that all the easier - the warm reboot vector would skip all the initial copying of exception vectors and whatnot (and would leave alone any RAM area we decide to reserve).